### PR TITLE
[Refactor] to_bitmap(-1) return NULL instead of return parse failed error_message

### LINKED ISSUE #8357: [Refactor] to_bitmap(-1) return NULL instead of return parse failed error_message
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

ALWAYS_NULLABLE:

	to_bitmap（-1） return NULL；
	to_bitmap（NULL）return NULL；

	bitmap_from_string('-1') return NULL;
	bitmap_from_string(NULL) return NULL;

ALWAYS_NOT_NULLABLE:

	bitmap_count(NULL)  return 0;
	bitmap_hash(NULL)   return empty bitmap
	bitmap_union(NULL)  return empty bitmap        
	bitmap_intersect(NULL)    return empty bitmap

ALWAYS_NOT_NULLABLE

	hll_hash(NULL)  return empty hll

	hll_cardinality(NULL) return 0;

	hll_union_agg(NULL)   return 0;

	hll_union(NULL)  return empty hll

### Solution

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhangstar333**
# Proposed changes

Issue Number: close #8357 

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/functions/function_bitmap.cpp [hunk 1]
```diff
@@ -125,23 +151,48 @@ struct BitmapHash {
...
+        if (auto* nullable = check_and_get_column<const ColumnNullable>(*column)) {
+            VectorizedUtils::update_null_map(null_map, nullable->get_null_map_data());
+            column = nullable->get_nested_column_ptr();
+        }
+        auto str_col = assert_cast<const ColumnBitmap*>(column.get());
+        const auto& col_data = str_col->get_data();
+
+        res.reserve(input_rows_count);
+        for (size_t i = 0; i < input_rows_count; ++i) {
+            if(null_map[i]) {
```
Comment (HappenLee) [inline be/src/vec/functions/function_bitmap.cpp:189]: if (

File: be/src/vec/functions/function_bitmap.cpp [hunk 2]
```diff
@@ -125,23 +151,48 @@ struct BitmapHash {
     }
 };
 
-struct NameBitmapCount {
+class FunctionBitmapCount : public IFunction {
```
Comment (HappenLee) [inline be/src/vec/functions/function_bitmap.cpp:154]: why bitmap count is always nullable?

File: be/src/vec/aggregate_functions/aggregate_function_hll_union_agg.h [hunk 3]
```diff
@@ -52,73 +51,115 @@ struct AggregateFunctionHLLData {
...
+    static const DataTypePtr return_type_union() {
+        if constexpr (is_nullable) {
+            return make_nullable(std::make_shared<DataTypeHLL>());
+        } else {
+            return std::make_shared<DataTypeHLL>();
+        }
+    }
+
+    void add(const IColumn* column, size_t row_num) {
+        if (auto* nullable_column = check_and_get_column<const ColumnNullable>(*column)) {
```
Comment (HappenLee) [inline be/src/vec/aggregate_functions/aggregate_function_hll_union_agg.h:96]: if constexpr ？

File: be/src/vec/aggregate_functions/aggregate_function_hll_union_agg.h [hunk 4]
```diff
@@ -52,73 +51,115 @@ struct AggregateFunctionHLLData {
...
+            assert_cast<ColumnInt64&>(null_column.get_nested_column())
+                    .get_data()
+                    .push_back(get_cardinality());
+        } else {
+            assert_cast<ColumnInt64&>(to).get_data().push_back(get_cardinality());
+        }
+    }
+
+    void result_union_impl(IColumn& to) const {
+        if constexpr (is_nullable) {
```
Comment (HappenLee) [inline be/src/vec/aggregate_functions/aggregate_function_hll_union_agg.h:69]: maybe the result is always not nullable

File: be/src/vec/aggregate_functions/aggregate_function_hll_union_agg.h [hunk 5]
```diff
@@ -52,73 +51,115 @@ struct AggregateFunctionHLLData {
...
+                    static_cast<const ColumnHLL&>((nullable_column->get_nested_column()));
+            dst_hll.merge(sources.get_element(row_num));
+        } else {
+            const auto& sources = static_cast<const ColumnHLL&>(*column);
+            dst_hll.merge(sources.get_element(row_num));
+        }
+    }
+};
+
+template <typename Data>
```
Comment (HappenLee) [inline be/src/vec/aggregate_functions/aggregate_function_hll_union_agg.h:110]: refactor the struct template

File: be/src/vec/sink/vtablet_sink.cpp [hunk 6]
```diff
@@ -174,7 +174,7 @@ Status VOlapTableSink::_validate_data(RuntimeState* state, vectorized::Block* bl
                 block->get_by_position(i).column->convert_to_full_column_if_const();
         const auto& column = block->get_by_position(i).column;
 
-        if (desc->is_nullable() && desc->type() == TYPE_OBJECT) {
+        if (desc->is_nullable() && desc->type() == TYPE_OBJECT && column->is_nullable()) {
```
Comment (BiteTheDDDDt) [inline be/src/vec/sink/vtablet_sink.cpp:177]: desc->is_nullable() can remove

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**